sympy/solvers/solveset.py

```diff
@@ -365,7 +365,10 @@ def _solve_as_rational(f, symbol, domain):
     f = together(f, deep=True)
     g, h = fraction(f)
     if not h.has(symbol):
-        return _solve_as_poly(g, symbol, domain)
+        try:
+            return _solve_as_poly(g, symbol, domain)
+        except NotImplementedError:
+            return ConditionSet(f, symbol, domain)
```

Comment:
This looks good. It might be good to add an inline comment indicating why this might fail and to make sure that we have coverage for both cases.